src/modules/musig/adaptor_impl.h

```diff
@@ -0,0 +1,84 @@
+/**********************************************************************
+ * Copyright (c) 2021 Jonas Nick                                      *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+#ifndef SECP256K1_MODULE_MUSIG_ADAPTOR_IMPL
+#define SECP256K1_MODULE_MUSIG_ADAPTOR_IMPL
+
+#include "session.h"
+
+int secp256k1_musig_nonce_parity(const secp256k1_context* ctx, int *nonce_parity, const secp256k1_musig_session *session) {
+    secp256k1_musig_session_internal session_i;
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(nonce_parity != NULL);
+    ARG_CHECK(session != NULL);
+
+    if (!secp256k1_musig_session_load(ctx, &session_i, session)) {
+        return 0;
+    }
+    *nonce_parity = session_i.fin_nonce_parity;
+    return 1;
+}
+
+int secp256k1_musig_adapt(const secp256k1_context* ctx, unsigned char *sig64, const unsigned char *sec_adaptor32, int nonce_parity) {
```

Comment:
Since this (also `extract_adaptor`) is public facing, is it worth adding some check like `ARG_CHECK(nonce_parity == 0 || nonce_parity == 1)`?

Reply:
Since the API docs say that the nonce_parity arg must be the output of the getter, I think that makse sense.